File: 11/zmq/publish.py
#!/usr/bin/env python
#
# Publisher program
#   Sends "Hello World" to subscriber

import time
import zmq

print("Connecting to 5555")
context = zmq.Context()
socket = context.socket(zmq.PUB)
socket.connect("tcp://localhost:5555")

#  Do 10 publish
for request in range(10):
    ss = "Hello World! {}".format(request)
    print("Sending: {}".format(ss))
    socket.send(ss.encode('utf-8'))

    time.sleep(1)

File: 11/zmq/publish.cpp
#include <iostream>
#include <string>
#include <chrono>
#include <thread>
#include <zmq.hpp>

int main(int argc, char *argv[]) {
  // initialize the 0MQ context
  zmq::context_t context(1);

  // generate a publish socket
  zmq::socket_t socket (context, zmq::socket_type::pub);

  // open the connection
  std::cout << "Connecting to 5555" << std::endl;
  socket.bind("tcp://*:5555");

  // PUB/SUBでは接続直後のメッセージが落ちることがあるため少し待つ
  std::this_thread::sleep_for(std::chrono::milliseconds(500));

  int request_nbr;
  for (request_nbr = 0; request_nbr != 10; request_nbr++) {
    // compose a message from a string and a number
    std::stringstream ss;
    ss << "Hello World " << request_nbr;
    std::cout << "Sending: " << ss.str() << std::endl;

    // send a message
    std::string text = ss.str();
    socket.send(zmq::buffer(text), zmq::send_flags::none);

    // wait for next loop
    std::this_thread::sleep_for(std::chrono::seconds(1));
  }
}
